drizzlepac/devutils/comparison_tools/read_hla/omegaxyz.py

```diff
@@ -0,0 +1,556 @@
+
+    Parameters
+    ----------
+    a : 3-element list of floats
+        first set of values in cross-product calculation
+
+    b : 3-element list of floats
+        second set of values in cross-product calculation
+
+    Returns
+    -------
+    c :  3-element list of floats
+        result of cross-product calculation
+    """
+    c = [0]*3
+    c[0] = a[1]*b[2] - a[2]*b[1]
+    c[1] = a[2]*b[0] - a[0]*b[2]
+    c[2] = a[0]*b[1] - a[1]*b[0]
+    return c
+
+
+def dotproduct(a, b):
+    """Return a . b
+
+    Parameters
+    ----------
+    a : 3-element list of floats
+        first set of values in dot-product calculation
+
+    b : 3-element list of floats
+        second set of values in dot-product calculation
+
+    Returns
+    -------
+    c :  3-element list of floats
+        result of dot-product calculation
+    """
+    return a[0]*b[0]+a[1]*b[1]+a[2]*b[2]
+
+
+def getdataset(fitsfile):
+    """Extract visit identifier from FITS filename
+    
+    Returns None if this does not look like an HLA dataset name
+    This works on either a plain dataset name (hst_10188_10_acs_wfc_f814) or a
+    getdata.cgi URL (http://hla.stsci.edu/getdata.cgi?dataset=hst_10888_10_acs_wfc_f814w)
+
+    Parameters
+    ----------
+    fitsfile : str
+        fits file name
+
+    Returns
+    -------
+    vis_id : str
+        visit identifier value. Returns logical 'None' if this does not look like an HLA dataset name
+    """
+
+    f = os.path.split(fitsfile)[-1].split('_')
+    if len(f) < 5 or f[0].lower() != 'hst':
+        # perhaps this is a getdata URL?
+        ff = fitsfile.split('dataset=')
+        f = os.path.split(ff[-1])[-1].split('_')
+        if len(f) < 5 or f[0].lower() != 'hst':
+            return None
+    if f[3].lower() == 'wfpc2':
+        # look for shift for either WFPC2 or PC using WFPC2 catalogs
+        f[4] = 'wfpc2'
+    vis_id = '_'.join(f[:5])
+    return vis_id
+
+
+def applyomegawcs(filename, crval, cdmatrix, omega=None):
+
+    """Get the omega values for this dataset and return the updated (crval, cdmatrix) values
+
+    Parameters
+    ----------
+    filename : str
+        dataset filename
+
+    crval : list of floats
+        Right Ascension and Declination position at the reference pixel
+
+    cdmatrix : list of floats
+        Description of linear distortions: plate scale, rotation, and skew of the image
+
+    omega : tuple, optional. Default value = None
+        3-element tuple containing omega X, Y, and Z components.
+
+    Returns
+    -------
+    Updated versions of input variables crval and omega
+    """
+    if (not crval) or (not cdmatrix):
+        return (crval, cdmatrix)
+
+    if not omega:
+        # parse the filename and extract the visit identification
+        dataset = getdataset(filename)
+        if not dataset:
+            return (crval, cdmatrix)
+        omega = getomegaxyz(dataset)
+
+    if omega[0] != 0 or omega[1] != 0 or omega[2] != 0:
+        d2r = math.pi/180
```

Comment:
For later...should use functions in math or numpy to convert degrees -> radians and radians -> degrees.

Reply:
Issue resolved in commit [1f8da23](https://github.com/spacetelescope/drizzlepac/pull/548/commits/1f8da234a95cf22f0b6f15e51e55ff8e98425d89)